ci-operator/step-registry/openshift/e2e/gcp/libvirt/cert-rotation/conf/openshift-e2e-gcp-libvirt-cert-rotation-conf-commands.sh

```diff
@@ -0,0 +1,428 @@
+  name: master
+  platform: {}
+  replicas: 3
+metadata:
+  creationTimestamp: null
+  name: installer
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 192.168.126.0/24
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  libvirt:
+    network:
+      if: tt0
+publish: External
+pullSecret: $(echo \'"${PULL_SECRET}"\')
+sshKey: |
+  $(cat "${PUB_SSH_KEY}")
+additionalTrustBundle: |
+  -----BEGIN CERTIFICATE-----
+  $(echo "${CA}")
+  -----END CERTIFICATE-----
+imageContentSources:
+- mirrors:
+  - ${HOSTNAME}:5000/ocp4/openshift4
+  source: quay.io/openshift-release-dev/ocp-release-nightly
+- mirrors:
+  - ${HOSTNAME}:5000/ocp4/openshift4
+  source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
+EOF
+
+# Create manifests and modify route domain
+openshift-install --dir="${CLUSTER_DIR}" create manifests
+# Workaround for https://github.com/openshift/installer/issues/1007
+# Add custom domain to cluster-ingress
+yq write --inplace $CLUSTER_DIR/manifests/cluster-ingress-02-config.yml spec[domain] apps.$BASE_DOMAIN
+
+# Add master memory to 14 GB
+# This is only valid for openshift 4.3 onwards
+yq write --inplace ${CLUSTER_DIR}/openshift/99_openshift-cluster-api_master-machines-0.yaml spec.providerSpec.value[domainMemory] 14336
+set -x
+openshift-install create cluster --log-level=debug --dir="$CLUSTER_DIR" || true
+openshift-install wait-for install-complete --log-level=debug --dir="$CLUSTER_DIR"
+END
+chmod +x "${SHARED_DIR}"/create-cluster-mirrored-local-registry
+
+cat  > "${SHARED_DIR}"/time-skew-test.sh << 'EOF'
+#!/bin/bash
+
+set -euxo pipefail
+
+approveCSRs () {
+  pendingCSRs=$(oc get csr | grep Pending | wc -l)
+  if [ $pendingCSRs -ne 0 ]; then
+    echo "Approving pending csrs"
+    oc get csr -o name | xargs oc adm certificate approve
+    sleep 30
+  fi
+}
+
+# TODO: Need to improve this nodesReady check
+checkNodesReady () {
+  nodesReady=0
+  retries=0
+  while [ $nodesReady -ne 5 ] && [ $retries -lt 30 ]; do
+    approveCSRs
+    nodesReady=$(oc wait --for=condition=Ready node --all --timeout=30s| wc -l)
+    if [ $nodesReady -eq 5 ]; then
+      echo "All nodes Ready"
+    fi
+    (( retries++ ))
+  done
+  if [ $nodesReady -ne 5 ]; then
+    echo "Some nodes NotReady"
+    oc get nodes
+    exit 1
+  fi
+}
+
+jumpstartNodes () {
+  approveCSRs
+  # jumpstart any stuck nodes, during recovery nodes will be rebooted
+  nodesDisabled=$(oc get nodes | grep "NotReady" | awk '{ print $1 }')
+  if [ ! -z "${nodesDisabled}" ]; then
+    nodeDisabledList=( $nodesDisabled )
+    for i in "${nodeDisabledList[@]}"
+    do
+      echo "Restarting stuck node ${i}..."
+      sudo virsh destroy "${i}"
+      sleep 30
+      sudo virsh start "${i}"
+      sleep 60
+    done
+    checkNodesReady
+  fi
+}
+
+checkDegradedCOs () {
+  retries=0
+  while ! oc wait co --all --for='condition=Degraded=False' --timeout=10s && [ $retries -lt 50 ]; do
```

Comment:
why is it better to retry it then raise the timeout?

Reply:
for this checkDegraded, the timeout alone will work, I'll update that.  For the checkAvailable, checkProgressing, I've found the jumpstartNodes is required to move the recovery along.